pkg/controller/elasticsearch/driver/fixtures.go

```diff
@@ -116,11 +116,12 @@ func newUpgradeTestPods(pods ...testPod) upgradeTestPods {
 	return result
 }
 
-func (u upgradeTestPods) toES(version string, maxUnavailable int) esv1.Elasticsearch {
+func (u upgradeTestPods) toES(version string, maxUnavailable int, annotations map[string]string) esv1.Elasticsearch {
```

Comment:
We could also get the disabled predicates from the Elasticsearch resource, using a receiver: it would avoid to manage this additional parameter (It would also move `DisableUpgradePredicatesAnnotation` closer to other annotations):

The idea would be to have something along those lines in `elasticsearch_types.go`:
```golang
func (es Elasticsearch) DisabledPredicates() set.StringSet {
	if es.Annotations == nil {
		return set.StringSet{}
	}
	disabledPredicates, exists := es.Annotations[DisableUpgradePredicatesAnnotation]
	if !exists {
		return set.StringSet{}
	}
	return set.Make(strings.Split(disabledPredicates, ",")...)
}
```

_(We could also create a "utility function" shared with `SuspendedPodNames` as I think we almost want the same result: a string set from an annotation)_

Also maybe add a relevant unit test with some "edge cases" in `elasticsearch_types_test.go`:

```golang
func TestElasticsearch_DisabledPredicates(t *testing.T) {
	tests := []struct {
		name string
		es   Elasticsearch
		want set.StringSet
	}{
		{
			name: "no annotations",
			es: Elasticsearch{ObjectMeta: metav1.ObjectMeta{
				Annotations: nil,
			}},
			want: set.StringSet{},
		},
		{
			name: "no annotation",
			es: Elasticsearch{ObjectMeta: metav1.ObjectMeta{
				Annotations: map[string]string{},
			}},
			want: set.StringSet{},
		},
		{
			name: "1 disabled predicate",
			es: Elasticsearch{ObjectMeta: metav1.ObjectMeta{
				Annotations: map[string]string{
					DisableUpgradePredicatesAnnotation: "foo",
				},
			}},
			want: set.Make("foo"),
		},
		{
			name: "2 disabled predicates",
			es: Elasticsearch{ObjectMeta: metav1.ObjectMeta{
				Annotations: map[string]string{
					DisableUpgradePredicatesAnnotation: "foo,bar",
				},
			}},
			want: set.Make("foo", "bar"),
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := tt.es.DisabledPredicates(); !reflect.DeepEqual(got, tt.want) {
				t.Errorf("Elasticsearch.DisabledPredicates() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

The logic in `runPredicates` would be:

```golang
func runPredicates(
	ctx PredicateContext,
	candidate corev1.Pod,
	deletedPods []corev1.Pod,
	annotations map[string]string,
	maxUnavailableReached bool,
) (*failedPredicate, error) {

	disabledPredicates := ctx.es.DisabledPredicates()
	for _, predicate := range predicates {
		canDelete, err := predicate.fn(ctx, candidate, deletedPods, maxUnavailableReached)
		if err != nil {
			return nil, err
		}
		if !canDelete {
			// if this specific predicate name is disabled by the disable predicate annotation
			// "eck.k8s.elastic.co/disable-upgrade-predicates", then ignore this predicate,
			// and continue processing the remaining predicates.
			if disabledPredicates.Has(predicate.name) || disabledPredicates.Has("*") {
				log.Info("Warning: disabling upgrade predicate because of annotation", "predicate", predicate.name, "namespace", ctx.es.Namespace, "es_name", ctx.es.Name)
				continue
			}
...
```

Reply:
this makes perfect sense, done.